File: 2020/day1.py
f = open('./data/day1.txt', 'r')
numbers = []
for line in f.readlines():
    numbers.append(int(line))

def firstAnswer(numbers):
    for x in numbers:
        for y in numbers:
            if x + y == 2020:
                print("First answer results:")
                print(f"\t{x} + {y} is 2020")
                print(f"\tanswer = {x * y}")
                return
firstAnswer(numbers)

def secondAnswer(numbers):
    for x in numbers:
        for y in numbers:
            for z in numbers:
                if x + y + z == 2020:
                    print("Second answer results:")
                    print(f"\t{x} + {y} + {z} is 2020")
                    print(f"\tanswer = {x * y * z}")
                    return
secondAnswer(numbers)

File: 2022/day1.php
<?php

$contents = file_get_contents(dirname(__FILE__) . '/data/day1.txt');
$data = explode("\n", $contents);

$elves = [];
$i = 0;

foreach ($data as $value) {
    if (empty($value)) {
        $i++;
        continue;
    }

    if (!array_key_exists($i, $elves)) {
        $elves[$i] = 0;
    }

    $elves[$i] += $value;
}

echo 'First answer: ' . max($elves) . PHP_EOL;

rsort($elves);

echo 'Second answer: ' . array_sum(array_slice($elves, 0, 3)) . PHP_EOL;